cmd/gencerts_test.go

```diff
@@ -0,0 +1,122 @@
+package cmd_test
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/OpenBazaar/openbazaar-go/cmd"
+	"github.com/OpenBazaar/openbazaar-go/schema"
+)
+
+func buildCertDirectory() (*string, func(), error) {
+	ctx := schema.SchemaContext{
+		DataPath:        schema.GenerateTempPath(),
+		TestModeEnabled: true,
+	}
+	certSchema := schema.MustNewCustomSchemaManager(ctx)
+	if err := certSchema.BuildSchemaDirectories(); err != nil {
+		return nil, nil, err
+	}
+	return &ctx.DataPath, certSchema.DestroySchemaDirectories, nil
+}
+
+func TestGenCertsGenericDefaults(t *testing.T) {
+	var dataPath, Destroy, schema_err = buildCertDirectory()
+	dataDir := string([]byte(*dataPath))
+	if schema_err != nil {
+		Destroy()
+		t.Fatal(schema_err)
+	}
```

Comment:
There's a way to make sure destroy() is included in every exit from this function. It's called `defer`. Once you know destroy is safe to call, you can include ` defer destroy()` and it will be executed just before the function exits.